src/rules/flake8_django/rules.rs

```diff
@@ -0,0 +1,225 @@
+                ReceiverDecoratorChecker,
+                Range::from_located(func),
+            ));
+        }
+        None
+    }
+}
+
+define_violation!(
+    pub struct ModelStringFieldNullable(pub String);
+);
+impl Violation for ModelStringFieldNullable {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let ModelStringFieldNullable(field) = self;
+        format!("Avoid using null=True on string-based fields such as {field}.")
+    }
+}
+const NOT_NULL_TRUE_FIELDS: [&str; 6] = [
+    "CharField",
+    "TextField",
+    "SlugField",
+    "EmailField",
+    "FilePathField",
+    "URLField",
+];
+impl ModelStringFieldNullable {
+    pub fn check(bases: &[Expr], body: &[Stmt]) -> Vec<Diagnostic> {
+        let mut errors = Vec::new();
+        if !bases.iter().any(is_model) {
+            return errors;
+        }
+        for statement in body.iter() {
+            let Assign {value, ..} = &statement.node else {
+                continue
+            };
+            if let Some(field_name) = ModelStringFieldNullable::check_nullable_field(value) {
+                errors.push(Diagnostic::new(
+                    ModelStringFieldNullable(field_name.to_string()),
+                    Range::from_located(value),
+                ));
+            }
+        }
+        errors
+    }
+
+    fn check_nullable_field(value: &Expr) -> Option<&str> {
+        let Call {func, keywords, ..} = &value.node else {
+           return None;
+        };
+        let valid_field_name = match &func.node {
+            Attribute { attr, .. } => {
+                if !NOT_NULL_TRUE_FIELDS.contains(&&**attr) {
+                    return None;
+                }
+                Some(attr)
+            }
+            Name { id, .. } => {
+                if !NOT_NULL_TRUE_FIELDS.contains(&&**id) {
+                    return None;
+                }
+                Some(id)
+            }
+            _ => None,
+        };
+        let Some(valid_field_name) = valid_field_name else {
+            return None;
+        };
+
+        let mut null_key = false;
+        let mut blank_key = false;
+        let mut unique_key = false;
+        for keyword in keywords.iter() {
+            let Constant {value: Bool(true), ..} = &keyword.node.value.node else {
+                continue
+            };
+            let Some(argument) = &keyword.node.arg else {
+                continue
+            };
+            match argument.as_str() {
+                "blank" => blank_key = true,
+                "null" => null_key = true,
+                "unique" => unique_key = true,
+                _ => continue,
+            }
+        }
+        if blank_key && unique_key {
+            return None;
+        }
+        if null_key {
+            return Some(valid_field_name);
+        }
+        None
+    }
+}
+fn is_model(base: &Expr) -> bool {
+    match &base.node {
+        Name { id, .. } => {
+            if id == "Model" {
```

Comment:
We should use `Checker` for this to ensure that `Model` is actually `django.models.Model` or whatever.

If you thread `checker: Checker` through here, you can do:

```rust
checker.resolve_call_path(&base).map_or(false, |call_path| call_path.as_slice() == ["django", "Model"])
```

That will ensure that we support things like:

```
from django import models

class Foo(models.Model):
  ...

from django.models import Model as Moo

class Foo(Moo):
  ...
```

...and so on. _And_ that we avoid flagging `Model` references that _aren't_ to `django.models.Model`.

Reply:
Just checking in -- are you able to address these refactors? Or do you want a hand?

Reply:
I didn't have time to fix those, i'll do it durning the weekend. Also good catch with resolving whole path to check whether it's the django model.

Reply:
Ok no worries! Was just trying to keep track of whether you planned to look into these changes or not.